### DISCUSSION TITLE
Why is the launch message logged as a warning?

### DISCUSSION THREAD

**Discussion by Obiwan1995** [rwf2/Rocket#2529] [Questions] [answered]
I installed the latest version (0.5.0-rc.3) et I saw that the launch message was logged as a warning: 

![image](https://user-images.githubusercontent.com/12659111/233672209-b925d551-4c45-41c0-b5aa-b1794fb0c1ee.png)

I believe it should be an info instead as all logs above are infos as well? Or is there a condition to display a warning?

**Comment by blackghost1987** [comment]
I think it was changed to make sure something is displayed with the default settings, which are not displaying INFO level stuff on a release build. There were multiple false bugreports complaining because the servert didn't output anything when starting up. On the other hand people were also complaining about this causing unnecessary alerts because it's treated as a WARN when nothing bad happened really. BTW Originally I think every launch message was displayed as WARN, and there were a PR to change them all the INFO here: https://github.com/SergioBenitez/Rocket/issues/1828

Then it got partially reverted in this commit, I can't find the related issues / discussions, but see the commit message for more info:
https://github.com/SergioBenitez/Rocket/commit/885cdfd61c399ec369714b510c901920490883a2

So I think the current behaviour is intended to be a "middle ground" between everything being WARN and everything being INFO. But it looks like you can't please everyone either way 🤷‍♂️

**Comment by Obiwan1995** [comment]
Maybe a middle ground would be to display everything in DEBUG except the launch log which would be in INFO. And only INFO logs would be displayed in release. Or maybe it's not that simple to do?

**Reply by SergioBenitez** [reply, marked as answer]
> Maybe a middle ground would be to display everything in DEBUG except the launch log which would be in INFO. And only INFO logs would be displayed in release. Or maybe it's not that simple to do?

There are many INFO level log statements that we don't want to emit during release, however, so this would be even more onerous for those who don't want this single log message.

I don't think there's any good solution as it stands. Presumably once we switch over to tracing, it'll be easier to have arbitrary log levels, and launch information can live in its own level.
